app/controllers/candidate_interface/adviser_sign_ups/interruptions_controller.rb

```diff
@@ -13,7 +13,9 @@ def update
         if @adviser_interruption_form.valid?
           @adviser_interruption_form.save
 
-          if @adviser_interruption_form.proceed_to_request_adviser?
+          if @adviser_interruption_form.proceed_to_request_adviser? && application_form&.prefill_preferred_teaching_subject_id
+            redirect_to candidate_interface_adviser_sign_up_path(application_form.id, preferred_teaching_subject_id: application_form.prefill_preferred_teaching_subject_id)
+          elsif @adviser_interruption_form.proceed_to_request_adviser?
```

Comment:
I think the `prefill_preferred_teaching_subject_id` method belongs on the `@adviser_interruption_form`. Then these lines can become something like:
```suggestion
          if @adviser_interruption_form.has_prefilled_preferred_teaching_subject?
            redirect_to candidate_interface_adviser_sign_up_path(application_form.id, preferred_teaching_subject_id: application_form.prefill_preferred_teaching_subject_id)
          elsif @adviser_interruption_form.proceed_to_request_adviser?
```

Reply:
💯